src/main/java/ch/njol/skript/expressions/ExprScripts.java

```diff
@@ -52,54 +49,43 @@
 	"\t\tsend \"Loaded Scripts: %enabled scripts%\" to player",
 	"\t\tsend \"Unloaded Scripts: %disabled scripts%\" to player"
 })
-@Since("2.5")
-public class ExprScripts extends SimpleExpression<String> {
+@Since("INSERT VERSION")
+public class ExprScripts extends SimpleExpression<Script> {
 
 	static {
-		Skript.registerExpression(ExprScripts.class, String.class, ExpressionType.SIMPLE,
-				"[all [of the]|the] scripts [1:without ([subdirectory] paths|parents)]",
-				"[all [of the]|the] (enabled|loaded) scripts [1:without ([subdirectory] paths|parents)]",
-				"[all [of the]|the] (disabled|unloaded) scripts [1:without ([subdirectory] paths|parents)]");
+		Skript.registerExpression(ExprScripts.class, Script.class, ExpressionType.SIMPLE,
+				"[all [[of] the]|the] scripts",
+				"[all [[of] the]|the] (enabled|loaded) scripts",
+				"[all [[of] the]|the] (disabled|unloaded) scripts");
 	}
 
-	private static final Path SCRIPTS_PATH = Skript.getInstance().getScriptsFolder().getAbsoluteFile().toPath();
-
 	private boolean includeEnabled;
 	private boolean includeDisabled;
-	private boolean noPaths;
+	private int pattern;
```

Comment:
You can get rid of this and use the two booleans above instead

Reply:
Other way around, actually, or we can't differentiate between just disabled scripts and all scripts :)

Reply:
Sure, one int is likely simpler. It might be worth including final int fields for mapping (e.g. ANY = 0, ENABLED = 1, etc)